### PR TITLE
Feat lnurl support

### PR DESCRIPTION
**Pull request by r1n04h**
> [!NOTE]
> **Medium Risk**
> Adds LNURL parsing/network calls and a new LNURL-withdraw flow that creates invoices and hits third-party callbacks; mistakes could misroute funds or break send/scan flows.
> 
> **Overview**
> Adds broader Lightning destination support by treating both LNURL and Lightning Addresses as LNURL-pay inputs (new `Lnurl.isLightningAddressOrLnurl`), updating send screens to fetch pay-service metadata for either format and clarifying UI/error messaging.
> 
> Introduces an **LNURL-withdraw** flow: QR scanning now asynchronously detects `withdrawRequest` LNURLs and routes to a new `withdraw-lightning` screen that loads the withdraw request, validates min/max (including msat→sat edge cases), creates an invoice on the selected Lightning layer, and calls the LNURL callback.
> 
> Enhances the send flow to preserve/display the original Lightning Address on the confirmation screen, and updates unit tests to cover the new helpers and async scan routing.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6b4d851fb9de4fd7dd09fb2ec32cb6fd93dd6089. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: shared/class/lnurl.ts [hunk 1]
```diff
@@ -335,4 +339,14 @@ export default class Lnurl {
...
     return !!splitted[0].trim() && !!splitted[1].trim();
   }
+
+  static async isLnurlWithdrawRequest(lnurl: string): Promise<boolean> {
+    const url = Lnurl.getUrlFromLnurl(lnurl);
+    if (!url) throw new Error('Invalid LNURL');
+
+    const reply = await fetch(url, { method: 'GET' });
+    const payload = (await reply.json()) as LnurlPayServiceBolt11Payload;
+
```
Comment (JohnnySilverhandBot) [inline shared/class/lnurl.ts:349]: You’re treating this like a typed pay payload, but LNURL-withdraw replies are different and can also come back with HTTP errors or {status: 'ERROR', reason}. At least check reply.status and handle ERROR before trusting tag, otherwise routing becomes roulette on flaky servers.

File: mobile/app/send/withdraw-lightning.tsx [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    setError('');
+
+    try {
+      const wallet = await BackgroundExecutor.lazyInitWallet(selectedLayer, accountNumber);
+      if (!walletSupportsLightning(wallet)) {
+        throw new Error('Selected network does not support Lightning withdrawals');
+      }
+
+      const amountSats = getWithdrawAmountSats(payload);
+      const invoiceResponse = await wallet.createLightningInvoice(amountSats, payload.defaultDescription || 'Layerzwallet LNURL withdraw');
```
Comment (JohnnySilverhandBot) [inline mobile/app/send/withdraw-lightning.tsx:188]: This always withdraws maxWithdrawable. If min != max, that’s a surprise spend the UI never asked consent for. Either force min==max, let the user pick an amount, or default to min to avoid ‘oops I withdrew the max’ support tickets.

**Review by JohnnySilverhandBot — COMMENTED** [review]
LNURL support is a good add, but a couple edge cases here will bite users the moment a provider doesn’t behave like your happy-path tests. Fix the assumptions, then we can pretend this is ‘done’.

File: mobile/src/modules/scan-routing.ts [hunk 3]
```diff
@@ -160,11 +162,24 @@ export function parseQrIntent(rawInput: string): QrIntent {
...
 
   switch (intent.type) {
     case 'lightning': {
+      try {
+        if (await Lnurl.isLnurlWithdrawRequest(intent.invoice)) {
+          const params: WithdrawLightningParams = {
+            lnurl: intent.invoice,
+            network: NETWORK_LIGHTNING,
+          };
+          router.push({
```
Comment (tomheisenberg1) [inline mobile/src/modules/scan-routing.ts:176]: I'll open a PR changing this to the expo router linking prop. I'll approve given that we're not using the prop atm

**Review by tomheisenberg1 — APPROVED** [review]

File: mobile/app/send/withdraw-lightning.tsx (hunk 2 above)
Comment (evalthis) [reply mobile/app/send/withdraw-lightning.tsx:188]: I agree, this is inconvenient

**Review by evalthis — CHANGES_REQUESTED** [review]
ln url withdraw should allow me to choose the amount
I'm trying to withdraw from lnmarkets, but I can't because amount has milisats and app throws me an error

Image

Also I'm not sure about "Withdraw BTC" I would say "Receive BTC"

File: mobile/app/send/withdraw-lightning.tsx (hunk 2 above)
Comment (r1n04h) [reply mobile/app/send/withdraw-lightning.tsx:188]: i have not seen a use case where i have to withdraw less. if app wants to give me money, ill take it all.
but ok, in case of trading, maybe user wants to leave some on exchange, so probably need an input to let user choose

**Comment by r1n04h** [comment]
> ln url withdraw should allow me to choose the amount I'm trying to withdraw from lnmarkets, but I can't because amount has milisats and app throws me an error
> 
> Image
> Also I'm not sure about "Withdraw BTC" I would say "Receive BTC"

good catch with fucking millisats

**Comment by r1n04h** [comment]
@evalthis vibecoded the amount selector
